src/optimizer/filter_combiner.cpp

```diff
@@ -407,8 +408,58 @@ TableFilterSet FilterCombiner::GenerateTableScanFilters(vector<idx_t> &column_id
 			     constant_value.second[0].constant.type().InternalType() == PhysicalType::BOOL)) {
 				//! Here we check if these filters are column references
 				filter_exp = equivalence_map.find(constant_value.first);
+
+				// Check for a struct_extract filter
 				if (filter_exp->second.size() == 1 &&
-				    filter_exp->second[0].get().type == ExpressionType::BOUND_COLUMN_REF) {
+				    filter_exp->second[0].get().type == ExpressionType::BOUND_FUNCTION) {
+					auto &func = filter_exp->second[0].get().Cast<BoundFunctionExpression>();
+
+					// Only support one-level struct_extract for now
+					if (func.function.name == "struct_extract" &&
```

Comment:
Can we do this recursively in a loop so that we can handle recursive struct extracts as well?